app/models/rule_event.rb

```diff
@@ -23,4 +23,11 @@ def dump
       "details" => @details.map(&:dump),
     }
   end
+
+  def as_json
+    dump.merge({
+      name: @details.first.timestamp.to_s(:db),
+      error: @details.any? { _1.level == :error },
+    })
+  end
```

Comment:
Since the view is rendered in React, we need a way to pass the information to javascript-land. Unfortunately, `dump` isn't exactly what we want here since a lot of the info is calculated on the fly, not persisted.

Reply:
I'm good with a separation betwen UI and persistence.

Since this is view specific shouldn't this logic be in the controller? Even in js land?